src/compiler/moduleSpecifiers.ts

```diff
@@ -980,19 +1007,23 @@ function tryGetModuleNameFromExportsOrImports(options: CompilerOptions, host: Mo
                     return { moduleFileToTry: combinePaths(packageName, fragment) };
                 }
                 if (declarationFile && containsPath(pathOrPattern, declarationFile, ignoreCase)) {
-                    const fragment = getRelativePathFromDirectory(pathOrPattern, declarationFile, /*ignoreCase*/ false);
+                    const fragment = changeFullExtension(getRelativePathFromDirectory(pathOrPattern, declarationFile, /*ignoreCase*/ false), getJSExtensionForFile(declarationFile, options));

```

Comment:
shouldnt this be dependent of file case sensitvity?

Reply:
Sure looks like it!

Reply:
Either that, or the `containsPath` call should be hard-coded to `false`. `pathOrPattern` is a concatenation of a path and an `"exports"` string key, and Node’s resolution algorithm is case sensitive according to my reading. We may be doing this wrong all around.

Reply:
paths and case-sensitivity .. ugh..

Reply:
Oh wait, it’s an `"exports"` value, not a key. Yeah, I think the whole thing should use the FS case sensitivity.

Reply:
Oh, this possible bug only occurs under `MatchingMode.Directory`, which was deprecated in Node.js 14 and EOL in 17, so I’m not going to touch this.

Reply:
vfs is case insensitive by default. Dont you need to test it on "case sensitive" that its not offered or doesnt do some wierd case replacement.

Reply:
The test I added shows that a difference in case between the VFS and the paths written in package.json doesn’t prevent auto-imports from working, which is what’s most important IMO. Also, this PR doesn’t touch/change anything about case sensitivity. This `ignoreCase` argument was already there.